notebooks/working/5.0-mab-simulation-causal-graph.md

```diff
@@ -961,6 +961,115 @@ def FitAlternativeRegression(regressions, reg_types, data):
     return regression_results
 ```
 
+## Regressions for each causal graph 
+
+
+### Drive Alone 
+
+```python
+V_Drive_Alone.draw()
+```
+
+```python
+Drive_Alone_Df = bike_data_long[bike_data_long['mode_id']==1]
+Drive_Alone_Df.reset_index(drop=True,inplace=True)
+Drive_Alone_Reg = FitAlternativeRegression(regressions={1:('total_travel_distance','total_travel_cost'),
+                                                        2:('total_travel_distance','total_travel_time')},
+                                           reg_types={1:'linear',2:'linear'},
+                                           data = Drive_Alone_Df)
+```
+
+### Shared-2
+
+```python
+V_Shared_2.draw()
+```
+
+```python
+Shared_2_Df = bike_data_long[bike_data_long['mode_id']==2]
+Shared_2_Df.reset_index(drop=True,inplace=True)
+Shared_2_Reg = FitAlternativeRegression(regressions={1:('total_travel_distance','total_travel_cost'),
+                                                        2:('total_travel_distance','total_travel_time')},
+                                           reg_types={1:'linear',2:'linear'},
+                                           data = Shared_2_Df)
+```
+
+### Shared-3+
+
+```python
+V_Shared_3p.draw()
+```
+
+```python
+Shared_3p_Df = bike_data_long[bike_data_long['mode_id']==3]
+Shared_3p_Df.reset_index(drop=True,inplace=True)
+Shared_3p_Reg = FitAlternativeRegression(regressions={1:('total_travel_distance','total_travel_cost'),
```

Comment:
You might get to this in the subsequent commits, but am I wrong in thinking that your pseudocode is:

1. get distribution of nodes with no parents.
2. get distribution of nodes with parents (i.e. fit the regressions)
3. simulate nodes with parents
4. simulate availability
5. simulate nodes with no parents

But then your actual code does 1, 2, 4, 5?
Just wanted to make sure I haven't missed anything!

Reply:
Actually, the pseudocode should do the following:

1. get distribution of nodes with no parents.
2. get distribution of nodes with parents (i.e. fit the regressions)
3. simulate nodes with no parents
4. simulate nodes with parents
5. simulate availability
6. simulate choices.

I simulate nodes with parents between lines 1089-1293